diff -up Python-2.4.6/Modules/_bsddb.c.orig Python-2.4.6/Modules/_bsddb.c
--- Python-2.4.6/Modules/_bsddb.c.orig	2010-03-20 16:47:14.718678928 -0600
+++ Python-2.4.6/Modules/_bsddb.c	2010-03-20 16:48:10.013649515 -0600
@@ -4317,9 +4317,9 @@ DBTxn_prepare(DBTxnObject* self, PyObjec
     if (!PyArg_ParseTuple(args, "s#:prepare", &gid, &gid_size))
         return NULL;
 
-    if (gid_size != DB_XIDDATASIZE) {
+    if (gid_size != DB_GID_SIZE) {
         PyErr_SetString(PyExc_TypeError,
-                        "gid must be DB_XIDDATASIZE bytes long");
+                        "gid must be DB_GID_SIZE bytes long");
         return NULL;
     }
 
@@ -4892,7 +4892,6 @@ DL_EXPORT(void) init_bsddb(void)
     /* allow apps to be written using DB_RPCCLIENT on older BerkeleyDB */
     _addIntToDict(d, "DB_RPCCLIENT", DB_CLIENT);
 #endif
-    ADD_INT(d, DB_XA_CREATE);
 
     ADD_INT(d, DB_CREATE);
     ADD_INT(d, DB_NOMMAP);

